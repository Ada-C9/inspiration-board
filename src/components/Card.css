.card {
  background-color: #F4FF81;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;

  -moz-box-shadow:0px 8px 10px 2px rgba(33,33,33,1);
  -webkit-box-shadow: 0px 8px 10px 2px rgba(33,33,33,.7);
  box-shadow: 0px 8px 10px 2px rgba(33,33,33,.7);

}

.card:nth-of-type(4n+5) {
  -o-transform:rotate(-1.5deg);
  -webkit-transform:rotate(-1.5deg);
  -moz-transform:rotate(-1.5deg);
  position:relative;
  top:-2px;
  background:rgb(225, 204, 255);
}

.card:nth-of-type(3n+9){
  -o-transform:rotate(.5deg);
  -webkit-transform:rotate(.5deg);
  -moz-transform:rotate(.5deg);
  position:relative;
  top:5px;
  background:rgb(204, 255, 204);
}

.card:nth-of-type(4n+3){
  -o-transform:rotate(-1deg);
  -webkit-transform:rotate(-1deg);
  -moz-transform:rotate(-1deg);
  position:relative;
  top:5px;
  background: rgb(255, 220, 70);
}

.card:nth-of-type(4n+2){
  -o-transform:rotate(1deg);
  -webkit-transform:rotate(1deg);
  -moz-transform:rotate(1deg);
  position:relative;
  background: rgb(146, 228, 254);
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  max-height: 115px;
  text-overflow: ellipsis;
  overflow: hidden;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  align-self: start;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  cursor: pointer;
}

.card__update {
  align-self: end;
  grid-column: 3 / 3;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
  cursor: pointer;
}
