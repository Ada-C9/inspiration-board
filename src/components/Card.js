import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './Card.css';

class Card extends Component {
  render() {
    return (
      <div className="card">
        <div className="card__content">
          <div className="card__content-text">
          {this.props.quote}
          </div>
          <div className="card__content-emoji">
          {emoji.getUnicode(`${this.props.emoji}`)}
          </div>
          <button className="card__delete" onClick={ () => {
            this.props.removeCard(this.props.index)
          } }>
            delete
          </button>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  quote: PropTypes.string,
  emoji: PropTypes.string,
  removeCard: PropTypes.func,
  index: PropTypes.number,
};

export default Card;
