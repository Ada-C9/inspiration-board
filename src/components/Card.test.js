import React from 'react';
import Card from './Card';
//mount  will mount the component into the virtualDOM
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
});
