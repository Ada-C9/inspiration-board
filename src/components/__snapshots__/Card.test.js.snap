// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <div
      className="card__content-text"
    />
    <div
      className="card__content-emoji"
    />
    <button
      className="card__delete"
      onClick={[Function]}
    >
      delete
    </button>
  </div>
</div>
`;
